addons/godot-firebase/storage/storage.gd

```diff
@@ -24,268 +34,274 @@ var _content_length : int
-						continue
-					if not item_name.begins_with(task.ref.full_path):
-						continue
-					if task.action == StorageTask.TASK_LIST:
-						var dir_path : Array = item_name.split("/")
-						var slash_count : int = task.ref.full_path.count("/")
-						item_name = ""
-						for i in slash_count + 1:
-							item_name += dir_path[i]
-							if i != slash_count and slash_count != 0:
-								item_name += "/"
-						if item_name in items:
-							continue
-					
-					items.append(item_name)
-			task.data = items
-		
-		_:
-			task.data = JSON.parse(_response_data.get_string_from_utf8()).result
-	
-	task.finished = true
-	task.emit_signal("task_finished")
-	
-	if not _pending_tasks.empty():
-		var next_task : StorageTask = _pending_tasks.pop_front()
-		_process_request(next_task)
+    var task := _current_task
+    requesting = false
+    
+    task.result = result
+    task.response_code = _response_code
+    task.response_headers = _response_headers
+    
+    match task.action:
+        StorageTask.TASK_DOWNLOAD:
+            task.data = _response_data
+        
+        StorageTask.TASK_DELETE:
+            references.erase(task.ref.full_path)
+            task.ref.valid = false
+            if typeof(task.data) == TYPE_RAW_ARRAY:
+                task.data = null
+        
+        StorageTask.TASK_DOWNLOAD_URL:
+            var json : Dictionary = JSON.parse(_response_data.get_string_from_utf8()).result
+            if json and json.has("downloadTokens"):
+                task.data = _base_url + _get_file_url(task.ref.full_path) + "?alt=media&token=" + json.downloadTokens
+            else:
+                task.data = ""
+        
+        StorageTask.TASK_LIST, StorageTask.TASK_LIST_ALL:
+            var json : Dictionary = JSON.parse(_response_data.get_string_from_utf8()).result
+            var items := []
+            if json and json.has("items"):
+                for item in json.items:
+                    var item_name : String = item.name
+                    if item.bucket != bucket:
+                        continue
+                    if not item_name.begins_with(task.ref.full_path):
+                        continue
+                    if task.action == StorageTask.TASK_LIST:
+                        var dir_path : Array = item_name.split("/")
+                        var slash_count : int = task.ref.full_path.count("/")
+                        item_name = ""
+                        for i in slash_count + 1:
+                            item_name += dir_path[i]
+                            if i != slash_count and slash_count != 0:
+                                item_name += "/"
+                        if item_name in items:
+                            continue
+                    
+                    items.append(item_name)
+            task.data = items
+        
+        _:
+            task.data = JSON.parse(_response_data.get_string_from_utf8()).result
+    
+    task.finished = true
+    if task.data.has("error"):
```

Comment:
The data property might not always be a dictionary, so it may throw an actual error otherwise. You might need to check if the data is a dictionary first.
`typeof(task.data) == TYPE_DICTIONARY`